src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/StraxCoinviewRule.cs

```diff
@@ -106,5 +117,32 @@ protected override void AllowSpend(TxOut prevOut, Transaction tx)
 
             // Otherwise allow the spend (do nothing).
         }
+
+        /// <inheritdoc />
+        public override async Task RunAsync(RuleContext context)
+        {
+            if (this.logic != null && context.ValidationContext.BlockToValidate.Header is PosBlockHeader posHeader && posHeader.HasSmartContractFields)
```

Comment:
Are the PosBlockHeader checks only done because of the PowBlocks earlier in the chain?

Reply:
yes